File: frontend/web-delta/src/app/models/debt/debt.model.ts
import {BaseModel} from '../base/base.model';

export class DebtModel extends BaseModel {

  private _value: Number;
  private _date: Date;
  private _customer: Number;

    /**
     * Getter value
     * @return {Number}
     */
  public get value(): Number {
    return this._value;
  }

    /**
     * Getter date
     * @return {Date}
     */
  public get date(): Date {
    return this._date;
  }

    /**
     * Getter customer
     * @return {Number}
     */
  public get customer(): Number {
    return this._customer;
  }

    /**
     * Setter value
     * @param {Number} value
     */
  public set value(value: Number) {
    this._value = value;
  }

    /**
     * Setter date
     * @param {Date} value
     */
  public set date(value: Date) {
    this._date = value;
  }

    /**
     * Setter customer
     * @param {Number} value
     */
  public set customer(value: Number) {
    this._customer = value;
  }
}

File: backend/server_delta/server_delta_app/models/debt/debt_model.py
from django.db import models
from server_delta_app import models as server_delta_app_models, managers


class DebtModel(server_delta_app_models.BaseModel):
    """
        Model that represent the customer debt registered in app
    """
    class Meta:
        db_table = "debt"

    value = models.DecimalField(max_digits=10, decimal_places=2)
    date = models.DateField()
    customer_dossier = models.ForeignKey(server_delta_app_models.CustomerDossierModel, on_delete=models.CASCADE, related_name='debts')

    objects = managers.DebtManager()
